test/utils.sh

```diff
@@ -399,3 +399,58 @@ extract_related_images_from_bundle(){
   echo "${related_images}" | tr ' ' '\n'
 
 }
+
+# This function will be used by tasks in build-definitions
+# It returns a map of {source: [mirror1, mirror2]} for imageDigestMirrorSet yaml
+process_image_digest_mirror_set() {
+  local yaml_input="$1"
+  local pullspec_map="{"
+
+  if ! echo "${yaml_input}" | yq '.' &>/dev/null; then
+    echo "Invalid YAML input" >&2
+    exit 2
+  fi
+
+  for entry in $(yq --output-format=json '.spec.imageDigestMirrors' <<<"${yaml_input}" | jq -c '.[]'); do
+    source=$(echo "${entry}" | jq -r '.source')
+    mirrors_list=$(echo "${entry}" | jq -r '.mirrors | map("\"" + . + "\"") | join(",")')
+    pullspec_map+="\"${source}\":[${mirrors_list}],"
+  done
+
+  pullspec_map="${pullspec_map%,}}"
+
+  echo "${pullspec_map}"
+
+}
+
+# This function will be used by tasks in build-definitions
+# It replaces the image pullspec with the mirror and returns the modified pullspec
+# The image should be in `<image>:<tag>`, `<image>@<digest>` or `<image>:<tag>@<digest>` format
+replace_image_pullspec() {
```

Comment:
How is this intended to be used? How do we envision that this might be used eventually? This description feels generic enough that it might be beneficial to use in multiple use cases. I don't know if this is necessarily the best place to define it.

Reply:
any operator check that'll utilize the imageDigestMirrorSet or something similar will probably be using this function. There are a bunch of those in the pipeline, which is why I added it here instead of keeping it to my task

Reply:
I guess I was thinking that this might be helpful for build time as well, but I'll keep that separate as we don't have a good way to share build and test tooling.